src/api/app/controllers/webui/users/tokens/users_controller.rb

```diff
@@ -0,0 +1,33 @@
+class Webui::Users::Tokens::UsersController < Webui::WebuiController
+  before_action :set_token
+
+  def index
+    @users = @token.shared_among
+    # TODO: @groups =
+  end
+
+  def create
+    user = User.find_by(login: params[:userid])
+
+    if @token.shared_among.include?(user)
+      redirect_back(fallback_location: root_path, notice: "User #{user.login} is already associated to the token")
+    elsif @token.shared_among << user
+      redirect_back(fallback_location: root_path, success: "User #{user.login} has been added to the token successfully")
+    end
+  end
+
+  def destroy
+    user = User.find(params[:id])
+    if @token.shared_among.destroy(user)
```

Comment:
Uh, be careful. https://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-destroy

Reply:
This is not risky, it's just removing the entry in the join table. As you can see here:

```
[7] pry(main)> token.shared_among.destroy(user) 
  TRANSACTION (0.6ms)  BEGIN
  HABTM_SharedAmong Load (0.8ms)  SELECT `workflow_token_users`.* FROM `workflow_token_users` 
WHERE `workflow_token_users`.`token_id` = 1 AND `workflow_token_users`.`user_id` = 1
  Token::Workflow::HABTM_SharedAmong Destroy (0.3ms)  DELETE FROM `workflow_token_users` WHERE
 `workflow_token_users`.`token_id` = 1 AND `workflow_token_users`.`user_id` = 1
  TRANSACTION (1.5ms)  COMMIT
```

```
[14] pry(main)> token.shared_among.delete(user)                                               
  TRANSACTION (0.3ms)  BEGIN                                                                  
  Token::Workflow::HABTM_SharedAmong Destroy (0.3ms)  DELETE FROM `workflow_token_users` WHERE
 `workflow_token_users`.`token_id` = 1 AND `workflow_token_users`.`user_id` = 1               
  TRANSACTION (1.4ms)  COMMIT
  ```
  
  And as it's explained here:
  https://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html

> What gets deleted?

> There is a potential pitfall here: [has_and_belongs_to_many](https://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_and_belongs_to_many) and [has_many](https://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) :through associations have records in join tables, as well as the associated records. So when we call one of these deletion methods, what exactly should be deleted?

> The answer is that it is assumed that deletion on an association is about removing the link between the owner and the associated object(s), rather than necessarily the associated objects themselves. So with [has_and_belongs_to_many](https://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_and_belongs_to_many) and [has_many](https://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) :through, the join records will be deleted, but the associated records won't.

> This makes sense if you think about it: if you were to call post.tags.delete(Tag.find_by(name: 'food')) you would want the 'food' tag to be unlinked from the post, rather than for the tag itself to be removed from the database.

Reply:
I'm simply going to replace `destroy` by `delete` as we don't need to fire the callbacks.